### DISCUSSION TITLE
Welcome to stk Discussions!

### DISCUSSION THREAD

**Discussion by lukasturcani** [lukasturcani/stk#261] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by NguyenThiHoaiLinh** [comment]
I am new to stk. Trying to follow the examples on the Documentation. Whenever "stk.ConstructedMolecule" is called, the 
AttributeError: type object 'Rotation' has no attribute 'from_matrix'
error happens.
What should I do to solve this?
-----------
import stk

bb1 = stk.BuildingBlock('NCCCN', [stk.PrimaryAminoFactory()])
bb2 = stk.BuildingBlock(
        smiles='O=CC(C=O)CC=O',
        functional_groups=[stk.AldehydeFactory()],
    )
tetrahedron = stk.cage.FourPlusSix((bb1, bb2))
cage1 = stk.ConstructedMolecule(tetrahedron)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-2183ad8f72d0> in <module>
      7     )
      8 tetrahedron = stk.cage.FourPlusSix((bb1, bb2))
----> 9 cage1 = stk.ConstructedMolecule(tetrahedron)

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/molecules/constructed_molecule.py in __init__(self, topology_graph)
     71         """
     72 
---> 73         construction_result = topology_graph.construct()
     74         super().__init__(
     75             atoms=construction_result.get_atoms(),

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/topology_graph/topology_graph.py in construct(self)
    394 
    395         state = self._get_construction_state()
--> 396         state = self._place_building_blocks(state)
    397         state = self._run_reactions(state)
    398         return ConstructionResult(state)

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/topology_graph/topology_graph.py in _place_building_blocks(self, state)
    449         """
    450 
--> 451         return self._implementation._place_building_blocks(state)
    452 
    453     def _run_reactions(self, state):

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/topology_graph/implementations/serial.py in _place_building_blocks(self, state)
     45                 edges=edges,
     46                 building_blocks=building_blocks,
---> 47                 results=placement_results,
     48             )
     49         return state

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/construction_state/construction_state.py in with_placement_results(self, vertices, edges, building_blocks, results)
    125             edges=edges,
    126             building_blocks=building_blocks,
--> 127             results=results,
    128         )
    129 

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/cage/cage_construction_state.py in _with_placement_results(self, vertices, edges, building_blocks, results)
     55     ):
     56         # Need to iterate multiple times through results.
---> 57         results = tuple(results)
     58         super()._with_placement_results(
     59             vertices=vertices,

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/topology_graph/implementations/serial.py in <lambda>(placement)
     38             )
     39             placement_results = map(
---> 40                 lambda placement: placement.get_result(),
     41                 placements,
     42             )

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/topology_graph/topology_graph/implementations/utilities.py in get_result(self)
     66         position_matrix = self._vertex.place_building_block(
     67             building_block=self._building_block,
---> 68             edges=self._edges,
     69         )
     70         position_matrix.setflags(write=False)

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/topology_graphs/cage/vertices.py in place_building_block(self, building_block, edges)
    239             ),
    240             target=edge_normal,
--> 241             origin=self._position,
    242         )
    243         fg_bonder_centroid = building_block.get_centroid(

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/molecules/molecule/molecule.py in with_rotation_between_vectors(self, start, target, origin)
    260             start=start,
    261             target=target,
--> 262             origin=origin,
    263         )
    264 

~/opt/anaconda3/lib/python3.7/site-packages/stk/molecular/molecules/molecule/molecule.py in _with_rotation_between_vectors(self, start, target, origin)
    156         # Set the origin of the rotation to "origin".
    157         self._with_displacement(-origin)
--> 158         rot_mat = rotation_matrix(start, target)
    159 
    160         # Apply the rotation matrix to the atomic positions to yield

~/opt/anaconda3/lib/python3.7/site-packages/stk/utilities/utilities.py in rotation_matrix(vector1, vector2)
    547     # matrix and allows for returns as quaternion or alternative
    548     # type in the future.
--> 549     return Rotation.from_matrix(
    550         i + vx + np.multiply(np.dot(vx, vx), mult_factor)
    551     ).as_matrix()

AttributeError: type object 'Rotation' has no attribute 'from_matrix'
​

**Reply by lukasturcani** [reply]
Hi there! What version of scipy are you using? `from_matrix` was added in version 1.4.0 so you may have to update.

Try running:

```sh
pip install scipy --ugprade
```

and let me know what you get when you run (before and after the update)

```sh

pip show scipy

```

For reference:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.from_matrix.html#scipy.spatial.transform.Rotation.from_matrix

**Reply by NguyenThiHoaiLinh** [reply]
Dear Lukas,
Thanks a lot for your quick response.
I updated scipy to 1.6.1 and now it works.

**Reply by lukasturcani** [reply]
Yay! Glad to hear :)
